larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArFormattingHelper.cc
+ *
+ *  @brief  Implementation of the formatting helper class.
+ *
+ *  $Log: $
+ */
+
+#include <iomanip>
+
+#include "Pandora/StatusCodes.h"
+
+#include "larpandoracontent/LArHelpers/LArFormattingHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+    void LArFormattingHelper::SetStyle(const LArFormattingHelper::Style &style, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(style), stream);
+    }
+   
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::SetColour(const LArFormattingHelper::Colour &colour, std::ostream &stream)
+    {
+        LArFormattingHelper::PrintFormatCharacter(static_cast<unsigned int>(colour), stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::ResetStyle(std::ostream &stream)
+    {
+        LArFormattingHelper::SetStyle(REGULAR, stream);
```

Comment:
Resetting the style is interesting... This is probably just fine for all our needs, but it's potentially possible for a user with very non-standard style setup to call the LArFormattingHelper. The reset will presumably then reset everything to its own standard style, rather than the input style? Having said that, it would be ugly to have to cache the arguments upon initial usage... what if multiple callers, etc... Much cleaner in this approach.

Reply:
This is a good point. To the best of my knowledge, this will reset back to the default colour / style used in the terminal preferences, but I haven't tested this. 

Maybe we could add a switch to allow users to turn off styles / colours globally? I am thinking something like IsMonitoringEnabled. I guess this isn't the nicest solution but if it was annoying for someone with a non-standard setup it would be good to be able to turn it off.

Reply:
If this was in the SDK, we absolutely would need to address this head on, but it's in the algorithm library, for at monitoring for *development* purposes. As such, we can probably justify keeping things simple to cover 99.9+% of use cases!